src/SearchNamespace.js

```diff
@@ -0,0 +1,155 @@
+    trackColumns: {
+      album: '[data-col="album"]',
+      artist: '[data-col="artist"]',
+      duration: '[data-col="duration"]',
+      playCount: '[data-col="play-count"]',
+      title: '[data-col="title"]',
+    },
+  };
+
+  constructor(...args) {
+    super(...args);
+    this.path = findContextPath();
+    if (this.path) {
+      this._watchForSearches();
+    }
+
+    this.addMethod('getSearchText', this.getSearchText.bind(this));
+    this.addMethod('getResults', this.getResults.bind(this));
+    this.addMethod('isSearching', this.isSearching.bind(this));
+    this.addMethod('performSearch', this.performSearch.bind(this));
+    this.addMethod('playResult', this.playResult.bind(this));
+  }
+
+  _text(elem, def) {
+    if (elem) {
+      return elem.textContent.trim();
+    }
+    return def;
+  }
+
+  _watchForSearches() {
+    const that = this;
+    let searchChanged = 0;
+
+    window.addEventListener('hashchange', () => {
+      if (!/^#\/sr\//g.test(window.location.hash)) return;
+      this.emitter.emit('change:search-text', this.getSearchText());
+      searchChanged = 2;
+    });
+
+    window.APPCONTEXT[this.path[0]][this.path[1]][0].addEventListener('E', () => {
+      if (searchChanged > 0) {
+        searchChanged -= 1;
+        if (searchChanged === 0) {
+          // DEV: We need to wait for GPM's own hooks to finish before scanning the UI
+          //      If we push this function to the end of the execution queue, the render
+          //      will complete syncronously before calling
+          setTimeout(() => {
+            that.emitter.emit('change:search-results', that.getResults());
+          }, 0);
+        }
+      }
+    });
+  }
+
+  getSearchText() {
+    return document.querySelector(SearchNamespace.selectors.inputBox).value;
+  }
+
+  getResults() {
+    if (!this.isSearching()) {
+      throw new Error('Can\'t get search results when the user is not searching');
+    }
+
+    const albumElems = document.querySelectorAll(SearchNamespace.selectors.albumResults);
+    const albums = [];
+
+    Array.prototype.forEach.call(albumElems, (elem) => {
+      albums.push(new Album(
+        elem.getAttribute('data-id'),
+        elem.querySelector(SearchNamespace.selectors.cardTitle).textContent,
+        elem.querySelector(SearchNamespace.selectors.cardSubTitle).textContent,
+        // DEV: Remove trailing query params from image with path such as
+        //      https://lh3.googleusercontent.com/4Yht2ETGQNme6QgQi-imsOK788OEHEhldqgBjeR8hWi8YsUMbn_AY0c5COHB4wK5C3Hjiw-y3Q=w220-c-h220-e100
+        elem.querySelector('img').src.replace('=w220-c-h220-e100', '')
+      ));
+    });
+
+    const artistElems = document.querySelectorAll(SearchNamespace.selectors.artistResults);
+    const artists = [];
+
+    Array.prototype.forEach.call(artistElems, (elem) => {
+      let image = elem.querySelector('img');
+      if (image) {
+        // DEV: Remove trailing query params from image with path such as
```

Comment:
Those aren't query parameters since they aren't in the query string (i.e. content after `?`). They are parameters in the path =/